### PR TITLE
Add test framework for C++ codes (googletest)

### PR DESCRIPTION
**Pull request by yjaeseok**
We need a test framework for C++ codes.
So I add googletest package as submodule.
And then make some testcase for calculator

After this patch, we can try to test for C++ codes.

$ mkdir build
$ cd build
$ cmake ..
$ make
$ ./unit_tests

You can see test results like below:

[==========] Running 4 tests from 1 test case.
[----------] Global test environment set-up.
[----------] 4 tests from Calculator
[ RUN      ] Calculator.Add
[       OK ] Calculator.Add (0 ms)
[ RUN      ] Calculator.Sub
[       OK ] Calculator.Sub (0 ms)
[ RUN      ] Calculator.Mul
[       OK ] Calculator.Mul (0 ms)
[ RUN      ] Calculator.Div
[       OK ] Calculator.Div (0 ms)
[----------] 4 tests from Calculator (0 ms total)

[----------] Global test environment tear-down
[==========] 4 tests from 1 test case ran. (1 ms total)
[  PASSED  ] 4 tests.

ISSUE=#47

### CODE REVIEW

**Review by romandev — COMMENTED** [review]
Wow! Thank you very much @yjaeseok!
(Also, please update AUTHORS file in this patch)

How can I use this test?
I don't have cmake in my environment and we are using GYP instead of cmake.
If it's not easy to integrate GYP now, could you update dependent packages to README.md for all platforms.

(Also, I think the unit tests should run when users execute ```./bacardi test``` command)

**Comment by romandev** [comment]
Anyway very nice commit description. Your description is a best practice to me.

**Comment by yjaeseok** [comment]
Thank you for your comments!
As you mentions, I will try how to integrate gyp module CMake, and combine `./bacardi test `

**Comment by romandev** [comment]
Yeah, but this may be a big deal than I thought.
Therefore, I'm okay to merge this patch as-is. (including AUTHORS and minimum README.md update)
and then you can migrate cmake to GYP.
I recommend to split this large work into smaller patches. (I think)

**Comment by yjaeseok** [comment]
Okay, anyway, which platform do you use ? Windows? Mac? Linux? else?..

**Comment by romandev** [comment]
@yjaeseok I'm using Mac and Linux. (But we are supporting Windows as well :))

**Comment by yjaeseok** [comment]
I updated README.md files for indicating we need to setup CMake program for C++ unit testing.
(But this is just temporary patch as @romandev mention !! )

And If user has CMake program, user can C++ unit test `./bacardi test` as run command only

**Comment by yjaeseok** [comment]
hmm.. 
This patch operate only unix/linux system.
Because of Make command. 

I will check it soon.

**Comment by yjaeseok** [comment]
I will consider how to make `gyp` `make` command.

**Comment by yjaeseok** [comment]
Finally, even though we use cmake module for buliding, I got successful unit test result in Windows Platform. 
For later, i will try to change cmake to gyp for building system and add unit test result test command also.

**Comment by yjaeseok** [comment]
After demo, please review again @romandev , @hwanseung

**Comment by yjaeseok** [comment]
Oh... T.T
for Linux or Unix system, target library must be changed for google test code.

So I should update CMakeLists.txt file.. OTZ ..

**Comment by romandev** [comment]
Cheer up! It's not easy to support multi-platforms.
My PR is also blocked by Windows platform although it works well in Linux and Mac. :)

**Comment by yjaeseok** [comment]
Please review. I updated to support Linux and Windows System.

File: README.md
Comment (romandev) [inline README.md]: I thought that this is necessary but my thinking was changed.
Sorry for that..

It's a WIP feature, so, I think it's better to hide this description.
until this feature is enabled fully.
WDYT?

File: bootstrap/command/test
Comment (romandev) [inline bootstrap/command/test]: This project is not Chromium but we are following Chromium style and policy if possible.
In Chromium, we try to avoid unused code or unused comments if possible.
Once the code is merged, it requires some reasons to remove them.
So, IMHO, it's better to remove these comment lines. WDYT?

File: test/calculator.cc
Comment (romandev) [inline test/calculator.cc]: I discussed about this test with @hwanseung but we thought that this test might not be necessary.
Because the calculator example is already tested by Mocha with JS.
But anyway, we need an test example for unittest as well.
So, I think we might have two options as follows:
  - Remove this test and rewrite a new test we really need (e.g. unit test for NativeTypeTraits)
  - Keep this test as-is for now.

The first option is good to me but I also don't want this PR to be pending.
So, the second option is also good to me.

WDYT? @hwanseung @yjaeseok

File: README.md
Comment (yjaeseok) [reply README.md]: Okay, I agree your comment! After this feature is enabled fully. I will re-consider update README or not.

File: bootstrap/command/test
Comment (yjaeseok) [reply bootstrap/command/test]: Ah... Thanks to you, I came to know Chromium Style. I will remove this comments!

File: test/calculator.cc
Comment (yjaeseok) [reply test/calculator.cc]: I choose to remove this test and rewrite a new test we really need 
But I am having trouble adding new test code right now.

So I would like to add only testing framework at this PR.

Comment (romandev) [reply test/calculator.cc]: @yjaeseok okay, it's a good idea. (and Congratulation on the birth of junior Jake!)

Comment (yjaeseok) [reply test/calculator.cc]: But....... If I do this, this PR becomes for .gitmodules only. T.T

Comment (yjaeseok) [reply test/calculator.cc]: So I will make the new test ASAP!

**Comment by yjaeseok** [comment]
@romandev I update this PR, please re-review!
(My wife is sleeping now :+1: )

File: test/calculator.cc
Comment (romandev) [reply test/calculator.cc]: @yjaeseok Ah, sorry for confusing.
I meant that we might not need calculator test.
Your CMakeList.txt and main() is still needed I thought.
But you can do it in second patch, which do you prefer?

**Review by romandev — APPROVED** [review]
lgtm

**Pull request merged**
